spec/abstractops.html

```diff
@@ -148,6 +170,80 @@ <h1>ToTemporalRoundingIncrement ( _normalizedOptions_, _dividend_, _inclusive_ )
     </emu-alg>
   </emu-clause>
 
+  <emu-clause id="sec-temporal-tosecondsstringprecision" aoid="ToSecondsStringPrecision">
+    <h1>ToSecondsStringPrecision ( _normalizedOptions_ )</h1>
+    <p>
+      The abstract operation ToSecondsStringPrecision combines the values of the options `smallestUnit` and `fractionalSecondDigits` to yield a precision for printing minutes and seconds to a string, and a rounding unit and increment.
+      The precision may be an integer 0 through 9 signifying a number of digits after the decimal point in the seconds, the string *"minute"* signifying not to print seconds at all, or the string *"auto"* signifying to drop trailing zeroes after the decimal point.
+    </p>
+    <emu-alg>
+      1. Let _smallestUnit_ be ? GetOption(_normalizedOptions_, *"smallestUnit"*, *"string"*, « *"minute"*, *"second"*, *"millisecond"*, *"microsecond"*, *"nanosecond"*, *"minutes"*, *"seconds"*, *"milliseconds"*, *"microseconds"*, *"nanoseconds"* », *undefined*).
+      1. If the value of _smallestUnit_ is in the Plural column of <emu-xref href="#table-temporal-singular-and-plural-units"></emu-xref>, then
+        1. Set _smallestUnit_ to the corresponding Singular value of the same row.
+      1. If _smallestUnit_ is *"minute"*, then
+        1. Return the new Record {
+            [[Precision]]: *"minute"*,
+            [[Unit]]: *"minute"*,
+            [[Increment]]: 1
+          }.
+      1. If _smallestUnit_ is *"second"*, then
+        1. Return the new Record {
+            [[Precision]]: 0,
+            [[Unit]]: *"second"*,
+            [[Increment]]: 1
+          }.
+      1. If _smallestUnit_ is *"millisecond"*, then
+        1. Return the new Record {
+            [[Precision]]: 3,
+            [[Unit]]: *"millisecond"*,
+            [[Increment]]: 1
+          }.
+      1. If _smallestUnit_ is *"microsecond"*, then
+        1. Return the new Record {
+            [[Precision]]: 6,
+            [[Unit]]: *"microsecond"*,
+            [[Increment]]: 1
+          }.
+      1. If _smallestUnit_ is *"nanosecond"*, then
+        1. Return the new Record {
+            [[Precision]]: 9,
+            [[Unit]]: *"nanosecond"*,
+            [[Increment]]: 1
+          }.
+      1. Let _digits_ be ? GetStringOrNumberOption(_normalizedOptions_, *"fractionalSecondDigits"*, « *"auto"* », 0, 9, *"auto"*).
```

Comment:
Maybe it would have reduced my earlier confusion if there was a step above this saying "Assert: _smallestUnit_ is **undefined**."